### PR TITLE
Add FXIOS-9964 - Integrate password generator inside password manager script

### PR DESCRIPTION
**Pull request by issammani**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-9964)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/21883)

## :bulb: Description
This PR:
 - Integrates initial password generator inside password manager.
 - Fixes typo in `CC_Python_Update.py`
 - Adds API for swift and JS
   - `generatePassword` will be called by swift to get a password for the current domain.
   - `fillGeneratedPassword` will be called by swift to fill a password and confirmation field if it exists.
   - `postMessage({ type: "generatePassword"})` will be sent by JS to swift when a `new-password` field is detected.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] ~Wrote unit tests and/or ensured the tests suite is passing~
- [ ] ~When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)~
- [ ] ~If needed, I updated documentation / comments for complex code and public methods~
- [ ] ~If needed, added a backport comment (example `@Mergifyio backport release/v120`)~

### CODE REVIEW

File: firefox-ios/Client/Assets/CC_Script/CC_Python_Update.py [hunk 1]
```diff
@@ -42,20 +42,23 @@
     "toolkit/modules/third_party/fathom/fathom.mjs",
     "toolkit/components/passwordmgr/shared/LoginFormFactory.sys.mjs",
     "toolkit/components/passwordmgr/shared/NewPasswordModel.sys.mjs",
-    "toolkit/components/passwordmgr/shared/PasswordGenerator.sys.mjs ",
+    "toolkit/components/passwordmgr/shared/PasswordGenerator.sys.mjs",
```
Comment (issammani) [inline firefox-ios/Client/Assets/CC_Script/CC_Python_Update.py:45]: Space typo caused this file not be pulled 😅

File: firefox-ios/Client/Assets/CC_Script/CC_Python_Update.py [hunk 2]
```diff
@@ -67,16 +70,19 @@ def downloadTemporaryFileToCompare(file_info):
     except requests.exceptions.HTTPError as err:
         raise SystemExit(err)
 
+
```
Comment (nbhasin2) [inline firefox-ios/Client/Assets/CC_Script/CC_Python_Update.py:73]:
```suggestion
```
Q. do you need these spaces? Not sure if you added intentionally or it just got added

Comment (issammani) [reply firefox-ios/Client/Assets/CC_Script/CC_Python_Update.py:73]: Oh the formatter is being annoying. Let me fix that.

File: firefox-ios/Client/Frontend/UserContent/UserScripts/AllFrames/AtDocumentStart/LoginsHelper.js [hunk 3]
```diff
@@ -461,13 +491,23 @@ window.__firefox__.includeOnce("LoginsHelper", function() {
     if (!form) {
       return;
     }
-
+
```
Comment (nbhasin2) [inline firefox-ios/Client/Frontend/UserContent/UserScripts/AllFrames/AtDocumentStart/LoginsHelper.js:494]:
```suggestion
```
nit ^

File: firefox-ios/Client/Frontend/UserContent/UserScripts/AllFrames/AtDocumentStart/LoginsHelper.js [hunk 4]
```diff
@@ -444,6 +446,34 @@ window.__firefox__.includeOnce("LoginsHelper", function() {
     },
   }
 
+  const generatePassword = (rules) => {
+    let mapOfRules = null;
+
+    // This will be skipped for now, as we don't have the rules yet
+    // Once we have the rules, we can parse them and send an appropriate
+    // password to swift
+    if(rules) {
```
Comment (nbhasin2) [inline firefox-ios/Client/Frontend/UserContent/UserScripts/AllFrames/AtDocumentStart/LoginsHelper.js:455]: Are these the same rules we added in the app?

**Review by nbhasin2 — APPROVED** [review]
some questions / minor nits

File: firefox-ios/Client/Frontend/UserContent/UserScripts/AllFrames/AtDocumentStart/LoginsHelper.js (hunk 4 above)
Comment (issammani) [reply firefox-ios/Client/Frontend/UserContent/UserScripts/AllFrames/AtDocumentStart/LoginsHelper.js:455]: Yes it's the rules we added. This is an old comment. I will remove it. It should work now regardless if we have rules or not.

Comment (issammani) [reply firefox-ios/Client/Frontend/UserContent/UserScripts/AllFrames/AtDocumentStart/LoginsHelper.js:454]: Updated comment 😄

File: firefox-ios/Client/Frontend/UserContent/UserScripts/AllFrames/AtDocumentStart/LoginsHelper.js (hunk 3 above)
Comment (issammani) [reply firefox-ios/Client/Frontend/UserContent/UserScripts/AllFrames/AtDocumentStart/LoginsHelper.js]: Fixed !

**Pull request merged**